### PR TITLE
fix(controller): remove headers from model download api

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description
Unified the upload/download api for model and dataset:
1. dataset
changed param from 'part_name' to 'partName' 
2. model
changed headers from `'X-SW-DOWNLOAD-TYPE','X-SW-DOWNLOAD-OBJECT-NAME','X-SW-DOWNLOAD-OBJECT-PATH','X-SW-DOWNLOAD-OBJECT-HASH'` to `'desc','partName','path','signature' `and the type from header to param.
3. Change upload id from String to Long 

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Comment by jialeicui** [comment]
Should the upload interface match the download?

**Comment by goldenxinxing** [comment]
> Should the upload interface match the download?

done

File: server/controller/src/main/java/ai/starwhale/mlops/api/DatasetApi.java [hunk 1]
```diff
@@ -235,8 +232,8 @@ void pullDs(
             @PathVariable(name = "projectUrl") String projectUrl,
             @PathVariable(name = "datasetUrl") String datasetUrl,
             @PathVariable(name = "versionUrl") String versionUrl,
-            @Parameter(name = "part_name", description = "optional, _manifest.yaml is used if not specified")
-            @RequestParam(name = "part_name", required = false) String partName,
+            @Parameter(name = "signature", description = "optional, _manifest.yaml is used if not specified")
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/api/DatasetApi.java:235]: Not always signature is passed such as _manifest.yaml. Indeed , when it appears like a signature, it is a file who's name is the signature

File: server/controller/src/main/java/ai/starwhale/mlops/domain/model/mapper/ModelVersionMapper.java [hunk 2]
```diff
@@ -69,8 +69,8 @@ List<ModelVersionEntity> list(@Param("modelId") Long modelId,
     @Options(useGeneratedKeys = true, keyColumn = "id", keyProperty = "id")
     int insert(ModelVersionEntity version);
 
-    @Update("update model_version set status = #{status} where id = #{id}")
-    int updateStatus(@Param("id") Long id, @Param("status") Integer status);
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/model/mapper/ModelVersionMapper.java:73]: Could we gurantee that `versionName` will always be unique in the future especially crossing all projects. For example, after copy model cross projects.

Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/model/mapper/ModelVersionMapper.java]: fixed

File: server/controller/src/main/java/ai/starwhale/mlops/api/DatasetApi.java (hunk 1 above)
Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/api/DatasetApi.java]: fixed

**Review by anda-ren — APPROVED** [review]
LGTM

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**